contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -2062,3 +2067,290 @@ gen_sp_droprolemember_subcmds(const char *user, const char *member)
+				errmsg("input value is too long for function name")));
+	}
+	if(volatility != NULL)
+	{
+		/* strip trailing whitespace */
+		i = strlen(volatility);
+		while (i > 0 && isspace((unsigned char) volatility[i - 1]))
+			volatility[--i] = '\0';
+		
+		/* if volatility is empty */
+		if(i==0)
+			ereport(ERROR,
+					(errcode(ERRCODE_SYNTAX_ERROR),
+					errmsg("volatility is not valid")));
+
+		/* its length is greater than 9 (len of immutable) */
+		if (i > 9)
+			ereport(ERROR,
+				(errcode(ERRCODE_STRING_DATA_LENGTH_MISMATCH),
+				errmsg("input value is too long for volatility")));
+	}
+	if(function_name == NULL && volatility != NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
+				errmsg("function name cannot be NULL")));
+
+	if(function_name != NULL)
+	{	
+		List *function_name_list;
+		FuncCandidateList candidates = NULL;
+		char *full_function_name = NULL;
+		char *logical_schema_name = NULL;
+		char **splited_object_name;
+
+		/* get physical schema name */
+		splited_object_name = split_object_name(function_name);
+
+		if(strcmp(splited_object_name[0], "") || strcmp(splited_object_name[1], ""))
+			ereport(ERROR,
+						(errcode(ERRCODE_SYNTAX_ERROR),
+						errmsg("function \"%s\" is not a valid two part name", function_name)));
+		
+		pfree(function_name);
+		logical_schema_name = splited_object_name[2];
+		function_name = splited_object_name[3];
+		pfree(splited_object_name);
+		
+		/* downcase identifier */
+		if(pltsql_case_insensitive_identifiers)
+		{
+			logical_schema_name = downcase_identifier(logical_schema_name, strlen(logical_schema_name), false, false);
+			function_name = downcase_identifier(function_name, strlen(function_name), false, false);
+		}
+
+		/* truncate identifiers if needed */
+		tsql_schema_name = pstrdup(logical_schema_name);
+		truncate_tsql_identifier(logical_schema_name);
+		truncate_tsql_identifier(function_name);
+
+		if(!strcmp(function_name, ""))
+			ereport(ERROR,
+					(errcode(ERRCODE_SYNTAX_ERROR),
+					errmsg("function name is not valid")));
+
+		/* find the default schema for current user */
+		if (!strcmp(logical_schema_name, ""))
+		{	
+			const char *user = get_user_for_database(db_name);
+			const char *guest_role_name = get_guest_role_name(db_name);
+			if(!user)
+				ereport(ERROR,
+						(errcode(ERRCODE_UNDEFINED_OBJECT),
+						errmsg("user does not exist")));
+
+			pfree(logical_schema_name);
+			if ((guest_role_name && strcmp(user, guest_role_name) == 0))
+			{	
+				physical_schema_name = pstrdup(get_dbo_schema_name(db_name));
+				logical_schema_name = (char *)get_logical_schema_name(physical_schema_name, true);
+			}
+			else
+			{
+				logical_schema_name = get_authid_user_ext_schema_name((const char *) db_name, user);
+				physical_schema_name = get_physical_schema_name(db_name,logical_schema_name);
+			}
+		}
+		else
+		{
+			physical_schema_name = get_physical_schema_name(db_name,logical_schema_name);
+		}
+			
+		if(!strcmp(tsql_schema_name, ""))
```

Comment:
Why do we even require a separate variable for it tsql_schema_name? We are finding physical schema name and then deriving logical out of it, doesnt make sense to me